cmake/find-opensycl.py

```diff
@@ -0,0 +1,83 @@
+import os
+import sys
+from shutil import which
+import argparse
+import json
+
+
+def get_package_dir():
+  exe_name = which("syclcc")
+  bin_dir = os.path.dirname(exe_name)
+  return os.path.dirname(bin_dir)
+
+
+def get_config():
+  package_dir = get_package_dir()
+  etc_dir = os.path.join(package_dir, "etc")
+  config_file = None
+  for root, dirs, files in os.walk(etc_dir):
+    for file in files:
+      if file == "syclcc.json":
+        config_file = os.path.join(root, file)
+        break
+  
+  if not config_file:
+    print("failed to find opensycl config file", file=sys.stderr)
+    sys.exit(1)
+  
+  with open(config_file, 'r') as file:
+    data = json.load(file)
+  return data
+
+
+def get_full_target_name(arch):
+  data = get_config()
```

Comment:
Right now, `get_config` is getting called twice—but the result is the same. While that shouldn't cause many problems (I hope)... Wouldn't it be better only load the config once and then re-use it?